app_dart/test/service/github_checks_service_test.dart

```diff
@@ -0,0 +1,195 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:cocoon_service/src/model/luci/buildbucket.dart';
+import 'package:cocoon_service/src/model/luci/push_message.dart'
+    as push_message;
+import 'package:cocoon_service/src/service/github_checks_service.dart';
+import 'package:github/github.dart' as github;
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../model/github/checks_test_data.dart';
+import '../src/datastore/fake_cocoon_config.dart';
+import '../src/utilities/mocks.dart' as mocks;
+import '../src/utilities/mocks.dart';
+
+void main() {
+  FakeConfig config;
+  mocks.MockLuciBuildService mockLuciBuildService;
+  MockGitHub mockGitHub;
+  MockGithubChecksUtil mockGithubChecksUtil;
+  GithubChecksService githubChecksService;
+  github.CheckRun checkRun;
+
+  const Build linuxBuild = Build(
+    id: 998,
+    builderId: BuilderId(
+      project: 'flutter',
+      bucket: 'prod',
+      builder: 'Linux',
+    ),
+    status: Status.failure,
+  );
+
+  setUp(() {
+    config = FakeConfig();
+    mockGithubChecksUtil = MockGithubChecksUtil();
+    githubChecksService = GithubChecksService(
+      config,
+      githubChecksUtil: mockGithubChecksUtil,
+    );
+    mockLuciBuildService = mocks.MockLuciBuildService();
+    mockGitHub = MockGitHub();
+    config.githubClient = mockGitHub;
+    checkRun = github.CheckRun.fromJson(
+      jsonDecode(
+              '{"name": "Linux", "id": 123, "external_id": "678", "status": "completed", "started_at": "2020-05-10T02:49:31Z", "head_sha": "the_sha", "check_suite": {"id": 456}}')
+          as Map<String, dynamic>,
+    );
+    final Map<String, github.CheckRun> checkRuns = <String, github.CheckRun>{
+      'Linux': checkRun
+    };
+    when(mockGithubChecksUtil.allCheckRuns(any, any)).thenAnswer((_) async {
+      return checkRuns;
+    });
+  });
+
+  group('handleCheckSuiteEvent', () {
+    test('requested triggers all builds', () async {
+      await githubChecksService.handleCheckSuite(
+          checkSuiteString, mockLuciBuildService);
+      expect(
```

Comment:
Add an `expect` for multiple builds?

Reply:
This class does not have knowledge of the builds that need to get triggered. It delegates the execution of the builds to LuciBuildService and this use case is validated there.